src/nvim/api/ui_events.in.h

```diff
@@ -58,7 +58,8 @@ void set_title(String title)
   FUNC_API_SINCE(3);
 void set_icon(String icon)
   FUNC_API_SINCE(3);
-
+void highlight_info_set(Array highlights_changed)
+  FUNC_API_SINCE(3);
```

Comment:
I'm not against the "highlight changed" event, it will be very useful for other purposes. I'm just saying whenever we implement cursor colors in TUI (in this or next PR doesn't matter) we should do it the right way directly, so there will never be a need for TUI to receive this event.